plugins/database/src/db_plugin.cpp

```diff
@@ -13836,13 +13836,283 @@ irods::error db_get_hierarchy_for_resc_op(
+
+                if (std::strcmp(_arg4, "group") == 0) {
+                    // Return an error if the group does not exist.
+                    char user_id_string[MAX_NAME_LEN];
+                    auto ec = icatGetTicketGroupId(_ctx.prop_map(), _arg5, user_id_string);
+                    if (0 != ec) {
+                        return ERROR(ec, "icatGetTicketGroupId failed");
+                    }
+
+                    cllBindVars[0] = ticket_id_string.c_str();
+                    cllBindVars[1] = _arg5;
+                    cllBindVarCount = 2;
+
+                    return execute_sql("insert into R_TICKET_ALLOWED_GROUPS (ticket_id, group_name) values (? , ?)");
+                } // group
+            } // add
+            else if (std::strcmp(_arg3, "remove") == 0) {
+                if (std::strcmp(_arg4, "host") == 0) {
+                    // Return an error if the hostname cannot be converted to an IP address.
+                    char* hostIp = convertHostToIp(_arg5);
+                    if (!hostIp) {
+                        return ERROR(CAT_HOSTNAME_INVALID, "host name null");
+                    }
+
+                    cllBindVars[0] = ticket_id_string.c_str();
+                    cllBindVars[1] = hostIp;
+                    cllBindVarCount = 2;
+
+                    return execute_sql("delete from R_TICKET_ALLOWED_HOSTS where ticket_id = ? and host = ?");
+                } // host
+
+                if (std::strcmp(_arg4, "user") == 0) {
+                    // Return an error if the user does not exist.
+                    char user_id_string[MAX_NAME_LEN];
+                    auto ec = icatGetTicketUserId(_ctx.prop_map(), _arg5, user_id_string);
+                    if (0 != ec) {
+                        return ERROR(ec, "icatGetTicketUserId failed");
+                    }
+
+                    cllBindVars[0] = ticket_id_string.c_str();
+                    cllBindVars[1] = _arg5;
+                    cllBindVarCount = 2;
+
+                    return execute_sql("delete from R_TICKET_ALLOWED_USERS where ticket_id = ? and user_name = ?");
+                } // user
+
+                if (std::strcmp(_arg4, "group") == 0) {
+                    // Return an error if the group does not exist.
+                    char group_id_string[MAX_NAME_LEN];
+                    auto ec = icatGetTicketGroupId(_ctx.prop_map(), _arg5, group_id_string);
+                    if (0 != ec) {
+                        return ERROR(ec, "icatGetTicketGroupId failed");
+                    }
+
+                    cllBindVars[0] = ticket_id_string.c_str();
+                    cllBindVars[1] = _arg5;
+                    cllBindVarCount = 2;
+
+                    return execute_sql("delete from R_TICKET_ALLOWED_GROUPS where ticket_id = ? and group_name = ?");
+                } // group
+            } // remove
+        } // modify ticket operation
+
+        return ERROR(CAT_INVALID_ARGUMENT, "invalid op name");
```

Comment:
consider adding file/line to this error, pretty generic otherwise

Reply:
the `ERROR` macro does that.

Reply:
i totally knew that.